### ISSUE TITLE
Boot Effect/Color Won't Play/Clear After Reboot (RPi)

### ISSUE THREAD

**Issue by RPKoch** [hyperion-project/hyperion.ng#919]
- [x] I confirm that this is an issue rather than a question.

## Bug report

On a fresh boot or reboot of a Raspberry Pi, the Boot Effect/Color doesn't play or clear.

#### Steps to reproduce

(The following steps assume Hyperion.ng is already installed and setup properly)

1. Make sure Hyperion is running.
1. Open the web configuration from a different computer and navigate to `Configuration > Effects`. Note the Boot Effect/Color.
1. Navigate to the `Remote Control`
1. Initiate a `sudo reboot` on the Raspberry Pi

Watch the `Remote Control` screen in the browser. The web configuration will drop the connection to the server while the Pi reboots. Once the Pi completes its startup, the server will reconnect.

#### What is expected?

The Raspberry Pi reboots, the Hyperion service starts and plays the Boot Effect/Color, then clears and goes to the next source.

#### What is actually happening?

The Raspberry Pi reboots, the Hyperion service starts, but the Boot Effect/Color does not play and the Boot Effect/Color remains in the source list as with a priority of 0.

#### System

Hyperion Server:

- Build: master (GitHub-d663a48/d4b4158-1595872836)
- Build time: Jul 28 2020 03:15:29
- Git Remote: https://github.com/hyperion-project/hyperion.ng.git
- Version: 2.0.0-alpha.7
- UI Lang: auto (BrowserLang: en-US)
- UI Access: advanced
- Avail Capt: dispmanx,v4l2,framebuffer,qt

Hyperion Server OS:

- Distribution: Raspbian GNU/Linux 10 (buster)
- Arch: arm
- Kernel: linux (5.4.51+ (WS: 32))
- Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.105 Safari/537.36

**Comment by Paulchen-Panther** [comment]
Which LED device do you use?

**Comment by RPKoch** [comment]
I'm using the SK6812 LEDs

**Comment by tpmodding** [comment]
@RPKoch with and arduino or with the spi pins?

**Comment by RPKoch** [comment]
I tried it both on a Raspberry Pi Zero and a Raspberry Pi 3 Model B connected through the GPIO 10 (MOSI) port (pin 19).

**Comment by m-seker** [comment]
Can you start with -d and share the logs ?

**Comment by RPKoch** [comment]
I should have done that sooner. It looks like the issue may be related to having USB Capture on, and not having a USB capture device connected to the Pi.

```
pi@raspberrypi:~ $ journalctl -u hyperiond@pi.service
-- Logs begin at Mon 2020-08-03 07:37:38 EDT, end at Mon 2020-08-03 07:39:12 EDT. --
Aug 03 07:38:00 raspberrypi systemd[1]: Started Hyperion ambient light systemd service  for user pi.
Aug 03 07:38:07 raspberrypi hyperiond[361]: 2020-08-03T07:38:07.701 hyperiond MAIN         : <INFO> Set user data path to '/home/pi/.hyperion'
Aug 03 07:38:09 raspberrypi hyperiond[361]: 2020-08-03T07:38:09.834 hyperiond DAEMON       : <DEBUG> PythonInit.cpp:45:PythonInit() | Initializing Python interpreter
Aug 03 07:38:12 raspberrypi hyperiond[361]: 2020-08-03T07:38:12.594 hyperiond SETTINGSMGR  : <DEBUG> SettingsManager.cpp:108:SettingsManager() | Settings database initialized
Aug 03 07:38:12 raspberrypi hyperiond[361]: 2020-08-03T07:38:12.609 hyperiond DAEMON       : <INFO> CEC handler created
Aug 03 07:38:12 raspberrypi hyperiond[361]: 2020-08-03T07:38:12.850 hyperiond EFFECTFILES  : <INFO> 39 effects loaded from directory :/effects/
Aug 03 07:38:12 raspberrypi hyperiond[361]: 2020-08-03T07:38:12.970 hyperiond EFFECTFILES  : <INFO> 22 effect schemas loaded from directory :/effects/schema/
Aug 03 07:38:12 raspberrypi hyperiond[361]: 2020-08-03T07:38:12.992 hyperiond EFFECTFILES  : <INFO> 0 effects loaded from directory /home/pi/.hyperion/custom-effects
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.155 hyperiond SETTINGSMGR  : <DEBUG> SettingsManager.cpp:108:SettingsManager() | Settings database initialized
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.252 hyperiond BLACKBORDER  : <DEBUG> BlackBorderProcessor.cpp:65:handleSettingsUpdate() | Set mode to: default
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.267 hyperiond COMPONENTREG : <DEBUG> ComponentRegister.cpp:36:setNewComponentState() | Blackborder detector: enabled
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.458 hyperiond DAEMON       : <INFO> set screen capture device to 'dispmanx'
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.497 hyperiond DISPMANXGRAB : <DEBUG> Grabber.cpp:31:setVideoMode() | Set videomode to 0
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.539 hyperiond COMPONENTREG : <DEBUG> ComponentRegister.cpp:36:setNewComponentState() | Smoothing: enabled
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.558 hyperiond DISPMANXGRAB : <INFO> Display opened with resolution: 720x480
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.580 hyperiond DISPMANXGRAB : <DEBUG> Grabber.cpp:91:setWidthHeight() | Set new width: 80, height: 45 for capture
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.559 hyperiond LEDDEVICE    : <INFO> Start LedDevice 'sk6812spi'.
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.588 hyperiond LEDDEVICE    : <DEBUG> LedDevice.cpp:133:init() | deviceConfig: [{"colorOrder":"grb","currentLedCount":244,"hardwareLedCount":1,"invert":false,"latchTime":0,"output":"/dev/spid
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.605 hyperiond LEDDEVICE    : <DEBUG> LedDevice.cpp:153:init() | Refresh interval = 1000ms
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.604 hyperiond DAEMON       : <INFO> DISPMANX frame grabber created
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.634 hyperiond LEDDEVICE    : <DEBUG> ProviderSpi.cpp:46:init() | _baudRate_Hz [3000000], _latchTime_ms [0]
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.645 hyperiond CEC          : <INFO> Starting CEC handler
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.664 hyperiond LEDDEVICE    : <DEBUG> ProviderSpi.cpp:47:init() | _spiDataInvert [0], _spiMode [0]
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.674 hyperiond V4L2:/DEV/VI : <DEBUG> Grabber.cpp:31:setVideoMode() | Set videomode to 0
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.698 hyperiond LEDDEVICE    : <DEBUG> LedDeviceSk6812SPI.cpp:45:init() | whiteAlgorithm : subtract_minimum
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.769 hyperiond FLATBUFCONN  : <INFO> Connecting to Hyperion: 127.0.0.1:19401
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.800 hyperiond V4L2:/DEV/VI : <DEBUG> Grabber.cpp:91:setWidthHeight() | Set new width: 1024, height: 768 for capture
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.831 hyperiond V4L2:/DEV/VI : <INFO> Signal threshold set to: {12, 12, 12}
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.862 hyperiond V4L2:/DEV/VI : <INFO> Signal detection is now disabled
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.884 hyperiond V4L2:/DEV/VI : <INFO> Signal detection area set to: 0.250000,0.250000 x 0.750000,0.750000
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.915 hyperiond DAEMON       : <DEBUG> hyperiond.cpp:527:handleSettingsUpdate() | V4L2 grabber created
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.926 hyperiond JSONSERVER   : <DEBUG> JsonServer.cpp:27:JsonServer() | Created instance
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.839 hyperiond EFFECTENGINE : <INFO> Run effect "Rainbow swirl fast" on channel 0
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.956 hyperiond JSONSERVER   : <INFO> Started on port 19444
Aug 03 07:38:14 raspberrypi hyperiond[361]: 2020-08-03T07:38:14.997 hyperiond FLATBUFSERVE : <INFO> Started on port 19400
Aug 03 07:38:15 raspberrypi hyperiond[361]: 2020-08-03T07:38:15.078 hyperiond PROTOSERVER  : <INFO> Started on port 19445
Aug 03 07:38:15 raspberrypi hyperiond[361]: 2020-08-03T07:38:15.111 hyperiond WEBSERVER    : <DEBUG> WebServer.cpp:32:initServer() | Initialize Webserver
Aug 03 07:38:15 raspberrypi hyperiond[361]: 2020-08-03T07:38:15.158 hyperiond WEBSERVER    : <DEBUG> WebServer.cpp:32:initServer() | Initialize Webserver
Aug 03 07:38:15 raspberrypi hyperiond[361]: 2020-08-03T07:38:15.187 hyperiond EFFECTENGINE : <DEBUG> EffectEngine.cpp:187:runEffectScript() | Start the effect: name [Rainbow swirl fast], smoothCfg [2]
Aug 03 07:38:15 raspberrypi hyperiond[361]: 2020-08-03T07:38:15.252 hyperiond HYPERION     : <DEBUG> PriorityMuxer.cpp:158:registerInput() | Register new input 'System/EFFECT' with priority 0 as inactive
Aug 03 07:38:15 raspberrypi hyperiond[361]: 2020-08-03T07:38:15.265 hyperiond HYPERION     : <INFO> Initial foreground effect 'Rainbow swirl fast' started
Aug 03 07:38:15 raspberrypi hyperiond[361]: 2020-08-03T07:38:15.365 hyperiond EFFECTENGINE : <INFO> Run effect "Rainbow swirl fast" on channel 254
Aug 03 07:38:15 raspberrypi hyperiond[361]: 2020-08-03T07:38:15.454 hyperiond EFFECTENGINE : <DEBUG> EffectEngine.cpp:187:runEffectScript() | Start the effect: name [Rainbow swirl fast], smoothCfg [2]
Aug 03 07:38:15 raspberrypi hyperiond[361]: 2020-08-03T07:38:15.524 hyperiond HYPERION     : <DEBUG> PriorityMuxer.cpp:158:registerInput() | Register new input 'System/EFFECT' with priority 254 as inactive
Aug 03 07:38:15 raspberrypi hyperiond[361]: 2020-08-03T07:38:15.565 hyperiond HYPERION     : <INFO> Inital background effect 'Rainbow swirl fast' started
Aug 03 07:38:15 raspberrypi hyperiond[361]: 2020-08-03T07:38:15.589 hyperiond HYPERION     : <DEBUG> PriorityMuxer.cpp:158:registerInput() | Register new input 'System/V4L' with priority 240 as inactive
Aug 03 07:38:15 raspberrypi hyperiond[361]: 2020-08-03T07:38:15.625 hyperiond COMPONENTREG : <DEBUG> ComponentRegister.cpp:36:setNewComponentState() | V4L capture device: enabled
Aug 03 07:38:15 raspberrypi hyperiond[361]: 2020-08-03T07:38:15.642 hyperiond BOBLIGHT     : <DEBUG> BoblightServer.cpp:28:BoblightServer() | Instance created
Aug 03 07:38:15 raspberrypi hyperiond[361]: 2020-08-03T07:38:15.762 hyperiond COMPONENTREG : <DEBUG> ComponentRegister.cpp:36:setNewComponentState() | LED device: enabled
Aug 03 07:38:15 raspberrypi hyperiond[361]: 2020-08-03T07:38:15.826 hyperiond V4L2:/DEV/VI : <INFO> configured v4l device: /dev/video0
Aug 03 07:38:15 raspberrypi hyperiond[361]: 2020-08-03T07:38:15.882 hyperiond V4L2:/DEV/VI : <ERROR> Throws error nr: Cannot identify '/dev/video0' error code 2, No such file or directory
Aug 03 07:38:15 raspberrypi hyperiond[361]: Throws error nr: Cannot identify '/dev/video0' error code 2, No such file or directory
Aug 03 07:38:15 raspberrypi hyperiond[361]: 2020-08-03T07:38:15.901 hyperiond HYPERION     : <INFO> Hyperion instance 'First LED Hardware instance' has been started
Aug 03 07:38:16 raspberrypi hyperiond[361]: 2020-08-03T07:38:16.004 hyperiond HYPERION     : <DEBUG> PriorityMuxer.cpp:333:setCurrentTime() | Set visible priority to 0
Aug 03 07:38:19 raspberrypi hyperiond[361]: 2020-08-03T07:38:19.986 hyperiond WEBSERVER    : <DEBUG> WebServer.cpp:90:handleSettingsUpdate() | Apply Webserver settings
Aug 03 07:38:19 raspberrypi hyperiond[361]: 2020-08-03T07:38:19.992 hyperiond WEBSERVER    : <DEBUG> WebServer.cpp:108:handleSettingsUpdate() | Set document root to: :/webconfig
Aug 03 07:38:20 raspberrypi hyperiond[361]: 2020-08-03T07:38:20.005 hyperiond WEBSERVER    : <INFO> Started on port 8090 name 'Hyperion Webserver'
Aug 03 07:38:20 raspberrypi hyperiond[361]: 2020-08-03T07:38:20.044 hyperiond WEBSERVER    : <DEBUG> WebServer.cpp:90:handleSettingsUpdate() | Apply Webserver settings
Aug 03 07:38:20 raspberrypi hyperiond[361]: 2020-08-03T07:38:20.207 hyperiond WEBSERVER    : <DEBUG> WebServer.cpp:108:handleSettingsUpdate() | Set document root to: :/webconfig
Aug 03 07:38:20 raspberrypi hyperiond[361]: 2020-08-03T07:38:20.697 hyperiond WEBSERVER    : <DEBUG> WebServer.cpp:174:handleSettingsUpdate() | Setup SSL certificate
Aug 03 07:38:20 raspberrypi hyperiond[361]: 2020-08-03T07:38:20.786 hyperiond WEBSERVER    : <DEBUG> WebServer.cpp:190:handleSettingsUpdate() | Setup private SSL key
Aug 03 07:38:20 raspberrypi hyperiond[361]: 2020-08-03T07:38:20.870 hyperiond WEBSERVER    : <INFO> Started on port 8092 name 'Hyperion Webserver'
Aug 03 07:38:26 raspberrypi hyperiond[361]: 2020-08-03T07:38:26.175 hyperiond HYPERION     : <DEBUG> PriorityMuxer.cpp:238:setInputImage() | Priority 254 is now active
Aug 03 07:38:29 raspberrypi hyperiond[361]: 2020-08-03T07:38:29.266 hyperiond CEC          : <INFO> Auto detecting CEC adapter
Aug 03 07:38:29 raspberrypi hyperiond[361]: 2020-08-03T07:38:29.284 hyperiond CEC          : <INFO> CEC Adapter:
Aug 03 07:38:29 raspberrypi hyperiond[361]: 2020-08-03T07:38:29.288 hyperiond CEC          : <INFO>         Name   : RPI
Aug 03 07:38:29 raspberrypi hyperiond[361]: 2020-08-03T07:38:29.319 hyperiond CEC          : <INFO>         Path   : Raspberry Pi
Aug 03 07:38:39 raspberrypi hyperiond[361]: 2020-08-03T07:38:39.408 hyperiond CEC          : <ERROR> Failed to open the CEC adaper on port RPI
Aug 03 07:38:39 raspberrypi hyperiond[361]: Failed to open the CEC adaper on port RPI
Aug 03 07:38:43 raspberrypi hyperiond[361]: 2020-08-03T07:38:43.425 hyperiond WEBSOCKET    : <DEBUG> WebSocketClient.cpp:30:WebSocketClient() | New connection from ::ffff:192.168.7.247
Aug 03 07:38:43 raspberrypi hyperiond[361]: 2020-08-03T07:38:43.464 hyperiond WEBSOCKET    : <DEBUG> JsonAPI.cpp:107:handleInstanceSwitch() | Client '::ffff:192.168.7.247' switch to Hyperion instance 0
```
I can't remember if, when I was encountering this before, there was or wasn't a USB capture device connected. I am noticing I'm not seeing the issue on the Raspberry Pi 3 Model B anymore. Either way, hopefully this helps.

**Comment by Paulchen-Panther** [comment]
How long is your booteffect set for?

**Comment by RPKoch** [comment]
It's set to whatever the default duration is. I believe 3000 ms. I don't have my system up and running at the moment, but I will verify later.

**Comment by Lord-Grey** [comment]
Should be fixed by #1199
